src/main/java/com/epam/aidial/cfg/service/ApplicationResourceService.java

```diff
@@ -0,0 +1,149 @@
+    }
+
+    @Override
+    public ApplicationMetadataDto getMetadata(ResourceMetadataRequest request) {
+        var recursive = request.isRecursive();
+        var nextToken = request.getNextToken();
+        var path = request.getPath() != null ? request.getPath() : BASE_PATH;
+        var limit = request.getLimit() != null ? request.getLimit() : applicationsMetadataDefaultLimit;
+        return applicationClient.getApplicationMetadata(path, recursive, nextToken, limit);
+    }
+
+    @Override
+    public void move(MoveResource moveResource) {
+        var moveResourceDto = resourceClientMapper.toMoveResourceDto(moveResource, APPLICATIONS_PREFIX);
+        resourceClient.move(moveResourceDto);
+    }
+
+    @Override
+    public ResourceType getResourceType() {
+        return ResourceType.APPLICATION;
+    }
+
+    public ApplicationResource getApplicationResource(String path) {
+        return fetchApplicationResource(path, null).model();
+    }
+
+    public DomainModelWithEtag<ApplicationResource> getApplicationResource(String path, String etag) {
+        return fetchApplicationResource(path, etag);
+    }
+
+    private DomainModelWithEtag<ApplicationResource> fetchApplicationResource(String path, String etag) {
+        var response = applicationClient.getApplicationResource(path, createIfNonMatchHeaders(etag));
+
+        var metadata = applicationClient.getApplicationMetadata(
+                path,
+                false,
+                null,
+                applicationsMetadataDefaultLimit
+        );
+
+        var applicationResource = applicationClientMapper.toApplicationResource(response.getBody(), metadata);
+        var currentEtag = response.getHeaders().getETag();
+        return new DomainModelWithEtag<>(applicationResource, currentEtag);
+    }
+
+    public DomainModelWithEtag<ApplicationResource> putApplicationResource(CreateApplicationResource createApplicationResource,
+                                                         boolean allowOverride,
+                                                         String etag) {
+        var applicationResourceDto = applicationClientMapper.toApplicationResourceDto(createApplicationResource);
+        var path = applicationClientMapper.toPath(createApplicationResource);
+        var headers = createHeadersForCreate(allowOverride, etag);
+        var applicationMetadata = applicationClient.putApplicationResource(path, applicationResourceDto, headers);
+        var applicationResource = applicationClientMapper.toApplicationResource(applicationResourceDto, applicationMetadata.getBody());
+        var currentEtag = applicationMetadata.getHeaders().getETag();
+        return new DomainModelWithEtag<>(applicationResource, currentEtag);
+    }
+
+    public DomainModelWithEtag<ApplicationResource> createApplicationResource(CreateApplicationResource createApplicationResource,
+                                                                              boolean allowOverride,
```

Comment:
do you need allowOverride&allowOverride for createApplicationResource method?

Reply:
Done